### ISSUE TITLE
Problem with logical definition for abnormality of the chorioretinal region

### ISSUE THREAD

**Issue by cmungall** [obophenotype/human-phenotype-ontology#1030]
The HP textual definition is very clear: you have an abnormality of the chorioretinal region if you have an abnormality of the choroid AND if you have an abnormality of the retina.

This is also clearly reflected in the hierarchy:

<img width="1369" alt="screen shot 2016-09-01 at 8 37 10 am" src="https://cloud.githubusercontent.com/assets/50745/18221040/3e434d4e-7170-11e6-8f82-036db3726344.png">

(As an aside, I wonder why uveal abnormalities are not classified in the vascular branch too?)

The only issue is with the OWL definition, which is in terms of [UBERON:0019207](http://purl.obolibrary.org/obo/UBERON_0019207) ! chorioretinal region

This is a super-structure that encompasses both choroid and retina as sub-structures

```
'has part' some 
    (dysplastic
     and ('inheres in part of' some 'chorioretinal region')
     and ('has modifier' some abnormal))
```

what this is saying that if __any__ part of the CR region is dysplastic (ie any or both of C or R), then you have CR dysplasia. This is in contrast to the text def that states that if you must have __both__ dysplasia in the C __and__ dysplasia in the R to qualify.

**Comment by cmungall** [comment]
Note that it only takes one axiom like this to start causing a chain of incorrect inferences, such as: https://github.com/obophenotype/human-phenotype-ontology/issues/1030

**Comment by cmungall** [comment]
The solution is to take all these owl definitions:

<img width="889" alt="screen shot 2016-09-03 at 1 10 58 am" src="https://cloud.githubusercontent.com/assets/50745/18221312/5280b12c-7173-11e6-87ff-9356ac1e7bd0.png">

and replace:

```
HP:CRQ EquivalentTo (has-part some Q and (inheres-in-part-of some CR))
```

with:

```
HP:CRQ EquivalentTo HP:CQ and HP:RQ
```

being sure to document this pattern.

The important take-home here is that __language is deceptive__. We should be particularly wary of portmanteu terms when crafting OWL definitions. We may know internally the meaning but it is easy to fail to communicate this into OWL.

As an example, contrast the meaning of chorioretinal here with tracheobronchial here:

<img width="187" alt="screen shot 2016-09-03 at 1 16 09 am" src="https://cloud.githubusercontent.com/assets/50745/18221364/06f46748-7174-11e6-8580-4d2e35ffa851.png">

**Comment by cmungall** [comment]
any thoughts on this @drseb ?

**Comment by drseb** [comment]
my first thought: lets forbid all logical defs in HP that use other HP classes. What is the advantage of those? imho we don't gain anything by those. 

but I'll have another round of thinking

**Comment by cmungall** [comment]
I agree this is very problematic when we attempt to use an HPO class as a filler on a relation like inheres-in etc (there were some examples of these way back, horrible)

The hypothesis is that we gain correct inferences. If I have a C abnormality and an R abnormality, do i have a CR abnormality? It could be argued that if these are distinct abnormalities then I do not truly have a CR abnormality (e.g. the abnormality is in some sense uniform over the CR). If this is the case then my hypothesis is false and the HP-pair-intersection DP is not correct.

At the very least we must be careful with parthood here. I think the immediate problem is fixed if we change the IIPO to II for all merelogical sum abnormalities (this prevents abnormality of R from leaking to abnormality of CR). Note this does *not* apply to tracheobranchial tree abnormalities which I believe is intended as a union and not a sum.

**Comment by drseb** [comment]
Is this solved @cmungall ? If not, do you have time for a short skype this week?

**Comment by cmungall** [comment]
Not solved. But if you agree with the DP perhaps @nicolevasilevsky could implement?

**Comment by cmungall** [comment]
Some alternative:

 1. Change inheres-in-part-of to inheres-in for the combined region. 
 2. Use a DP: (has-part some X) AND (has-part some Y)

The 2nd is equivalent to a direct intersection between two HPO classes.

**Comment by pnrobinson** [comment]
@cmungall @drseb Can we discuss this and come to a conclusion? I agree that inheres in part of is less accurate that (inheres in part of X) and (inheres in part of Y) to model chorioretinal.

**Comment by pnrobinson** [comment]
@cmungall @drseb @nicolevasilevsky I agree with Chris's interpretation: chorioretinal is both C and R.  Currently I do not see a term for choroid in uberon. Could we make one and alter the definition of Chorioretinal abnormality (HP:0000532) accordingly? The other terms could be changed accordingly. Was this the consensus?

**Comment by pnrobinson** [comment]
@cmungall @drseb Any update on this?

**Comment by pnrobinson** [comment]
@cmungall and @drseb 
I tried this out and I think it solves some of the unwanted equivalencies. Is this OK? There are a number ofanalogous terms such as hepatosplenomegaly that probably need to be changed in this way.  Please let me know if you see any problems with this, otherwsie I will start to implement it next week.
```
'has part' some 
    (morphology
     and ('inheres in part of' some retina)
     and ('has modifier' some abnormal))
and
'has part' some 
    (morphology
     and ('inheres in part of' some 'optic choroid')
     and ('has modifier' some abnormal))
```

**Comment by cmungall** [comment]
This looks fine

**Comment by balhoff** [comment]
@pnrobinson I think you will need to change this like the one above. That definition requires something  to be both a spleen and a liver at the same time. So it could instead be:

```
('has part' some 
    ('increased size'
     and ('inheres in' some spleen)
     and ('has modifier' some abnormal)))
and
('has part' some 
    ('increased size'
     and ('inheres in' some liver)
     and ('has modifier' some abnormal)))
```
